### ISSUE TITLE
Warn on `.strip("...")` with multi-character strings - should use PEP-616 `.removeprefix`/`.removesuffix` methods

### ISSUE THREAD

**Issue by Zac-HD** [PyCQA/flake8-bugbear#228]
```python
fname = "clippy.py"
stem = fname.strip(".py")
print(stem)  # 'cli' - surprise!
```

The behavior of Python's `str.strip()` method (and `lstrip` and `rstrip`) when passed a multi-character string literal is a classic bugbear, and described well in [the abstract and rationale for PEP-616](https://peps.python.org/pep-0616/#abstract)

I think we should warn on *all* calls to `{,l,r}strip` methods passed a multi-character string as the sole positional argument, and suggest that the user either use PEP-616 affix methods, or make their intent explicit with `.strip(set("..."))`.

I'd enable this warning even when we can't be sure that the instance is a string - it's that common, and I expect that the false-positive rate will be very low compared to the additional detections.

**Comment by cooperlees** [comment]
Agree with all this. Seems a great one to warn everyone about!

**Comment by Zac-HD** [comment]
> make their intent explicit with `.strip(set("..."))`

...except that this is a `TypeError`, which limits options for avoiding false-positives.

I think the best option would be to restrict the check to cases where the string-to-strip contains duplicate characters (not a set), or matches `\.[a-z]+` (suspicious filename, can easily rearrange to avoid warning).

**Comment by johnkangw** [comment]
As part of the mentored sprints it looks like a duplicate of b005. We think this issue should be closed as this functionality already exists.

**Comment by Zac-HD** [comment]
Thanks for investigating!  You're right; issue closed 👍
